### PR TITLE
HPCC-19288 Enhancements to jutil queryHPCCPKIKeyFiles()

### PR DESCRIPTION
**Pull request by RussWhitehead**
Renamed environment.conf key 'HPCCCertFile' to 'HPCCPublicKeyFile'
Refactored jutil to query this new key name, and to return to password
as encrypted. Also added support for HPCCCertificate specification

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [x] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@mckellyln  Please review

File: system/jlib/jutil.cpp [hunk 1]
```diff
@@ -2455,25 +2455,41 @@ jlib_decl bool querySecuritySettings(DAFSConnectCfg *_connectMethod,
...
             }
-            *_passPhrase = DAFSpassPhrase.str();
         }
 
         if (!dfsKeywords && (_certificate || _privateKey || _passPhrase))
         {
         //end of deprecated code
-            queryHPCCPKIKeyFiles(_certificate, _privateKey, _passPhrase);//use new keywords
+            const char *passPhrasePtr = nullptr;
+            queryHPCCPKIKeyFiles(_certificate, _privateKey, _passPhrase ? &passPhrasePtr : nullptr);//use new keywords
```
Comment (mckellyln) [inline system/jlib/jutil.cpp]: I think in this querySecuritySettings() call a certificate file is expected and used for SSL handshake and authentication.  So the overlap of querySecuritySettings() and queryHPCCPKIKeyFiles() is not exact.

**Comment by mckellyln** [comment]
@RussWhitehead a comment and question about compatibility of these HPCCPKIKeyFiles() changes with the existiing querySecuritySettings() call.

**Comment by RussWhitehead** [comment]
@mckellyln Please review new support for certificate file

File: system/jlib/jutil.cpp [hunk 2]
```diff
@@ -2455,25 +2455,41 @@ jlib_decl bool querySecuritySettings(DAFSConnectCfg *_connectMethod,
...
+            queryHPCCPKIKeyFiles(_certificate, nullptr, _privateKey, _passPhrase ? &passPhrasePtr : nullptr);//use new keywords
+            if (!isEmptyString(passPhrasePtr))
+            {
+                CriticalBlock b(DAFSpassPhraseCrit);
+                if (DAFSpassPhraseEnc.isEmpty())
+                    DAFSpassPhraseEnc.append(passPhrasePtr);//got encrypted pwd
+            }
+        }
+
+        if (_passPhrase)
```
Comment (mckellyln) [inline system/jlib/jutil.cpp]: Add a comment that legacy querySecuritySettings() returns decrypted passPhrase but new HPCCPKIKeyFiles() returns unaltered passPhrase - so when used here we need to decrypt for compatibility.

Comment (RussWhitehead) [reply system/jlib/jutil.cpp]: @mckellyln Comments added

File: system/jlib/jutil.cpp [hunk 3]
```diff
@@ -2455,25 +2456,41 @@ jlib_decl bool querySecuritySettings(DAFSConnectCfg *_connectMethod,
...
+        {
+            CriticalBlock b(DAFSpassPhraseCrit);
+            if (DAFSpassPhraseDec.isEmpty()  &&  !DAFSpassPhraseEnc.isEmpty())
+                decrypt(DAFSpassPhraseDec, DAFSpassPhraseEnc.str());
+            *_passPhrase = DAFSpassPhraseDec.str();//return ENCRYPTED pwd. Note the preferred queryHPCCPKIKeyFiles() method returns it encrypted
```
Comment (mckellyln) [inline system/jlib/jutil.cpp]: DECRYPTED

**Comment by mckellyln** [comment]
@RussWhitehead change comment to DECRYPTED and then looks good to merge

**Comment by RussWhitehead** [comment]
@richardkchapman Please merge

**Comment by mckellyln** [comment]
@RussWhitehead can you squash ?

**Comment by RussWhitehead** [comment]
@mckellyln Squashed

**Pull request merged**